src/QAT/qat/purr/compiler/instructions.py

```diff
@@ -337,28 +337,27 @@ def __init__(
         output_variable=None,
         existing_names: Set[str] = None,
         delay=None,
-        filter: Pulse = None,
+        filter: Union[Pulse, CustomPulse] = None
     ):
         super().__init__(channel.full_id())
         super(QuantumComponent, self).__init__(channel)
         self.time: float = time or 1.0e-6
         self.mode: AcquireMode = mode or AcquireMode.RAW
         self.delay = delay
-        self.filter: Pulse = filter
         self.output_variable = output_variable or self.generate_name(existing_names)
-
-        if filter is not None:
-            if not isinstance(filter, Pulse):
-                raise ValueError(
-                    "Filter on an acquire has to be a Pulse. Instead it's a "
-                    f"{type(filter)}"
-                )
-
-            if filter.duration != self.time:
-                raise ValueError(
-                    f"Filter duration '{filter.duration}' must be equal to Acquire "
-                    f"duration '{self.time}'."
-                )
+        self.filter: Union[Pulse, CustomPulse] = self._check_filter(filter)
+
+    def _check_filter(self, filter):
+        if filter:
+            for target in self.quantum_targets:
+                if isinstance(target, PulseChannel):
+                    dt = target.physical_channel.sample_time
+                    if not np.isclose(filter.duration, dt * (self.time // dt), atol=1e-12):
```

Comment:
why `dt * (self.time // dt)` ?
use the systematic function `calculate_duration`
(@norbert-deak another reason why this function need not be an instance method)

Reply:
@hamidelmaazouz not sure I understand your comment here, might be missing some context

Reply:
logic here basically computes duration based on the channel's sample time, if I remember correctly, it's the same logic that's been specified at `QuantumExecutionEngine.calculate_duration()`

Reply:
Ok I've made it a bit clearer, I was trying to recreate the round error but with np close properly set, its not needed. Also calculate duration isn't needed as duration is calculated inside the CustomPulse